acceptance_modelisation/base_acceptance_map_creator.py

```diff
@@ -327,24 +297,43 @@ def _create_base_computation_map(self, observations: Observation) -> Tuple[WcsND
+                exclusion_mask = np.zeros(count_map_obs.counts.data.shape[1:])
                 for i in range(len(time_interval) - 1):
-                    cut_obs = obs.select_time(Time([time_interval[i], time_interval[i + 1]]))
-                    count_map_obs, exclusion_mask = self._create_camera_map(cut_obs)
-
-                    exp_map_obs = MapDataset.create(geom=count_map_obs.geoms['geom'])
-                    exp_map_obs_total = MapDataset.create(geom=count_map_obs.geoms['geom'])
-                    exp_map_obs.counts.data = cut_obs.observation_live_time_duration.value
-                    exp_map_obs_total.counts.data = cut_obs.observation_live_time_duration.value
-
-                    for j in range(count_map_obs.counts.data.shape[0]):
-                        count_map_obs.counts.data[j, :, :] = count_map_obs.counts.data[j, :, :] * exclusion_mask
-                        exp_map_obs.counts.data[j, :, :] = exp_map_obs.counts.data[j, :, :] * exclusion_mask
-
-                    count_map_background.data += count_map_obs.counts.data
-                    exp_map_background.data += exp_map_obs.counts.data
-                    exp_map_background_total.data += exp_map_obs_total.counts.data
-                    livetime += cut_obs.observation_live_time_duration
+                    # Compute the exclusion region in camera frame for the average time
+                    dtime = time_interval[i + 1] - time_interval[i]
+                    time = time_interval[i] + dtime/2
+                    average_alt_az_frame = AltAz(obstime=time,
+                                                 location=obs.observatory_earth_location)
+                    average_alt_az_pointing = obs.get_pointing_icrs(time).transform_to(average_alt_az_frame)
+                    exclusion_region_camera_frame = self._transform_exclusion_region_to_camera_frame(
+                        average_alt_az_pointing)
+                    geom_image = self.geom.to_image()
+
+                    exclusion_mask_t = ~geom_image.region_mask(exclusion_region_camera_frame) if len(
+                        exclusion_region_camera_frame) > 0 else ~Map.from_geom(geom_image)
+                    exclusion_mask += exclusion_mask_t * (dtime).value
+                exclusion_mask *= 1 / (time_interval[-1] - time_interval[0]).value
+
+                for j in range(count_map_obs.counts.data.shape[0]):
+                    exp_map_obs.counts.data[j, :, :] = exp_map_obs.counts.data[j, :, :] * exclusion_mask
+
+                count_map_background.data += count_map_obs.counts.data
+                exp_map_background.data += exp_map_obs.counts.data
+                exp_map_background_total.data += exp_map_obs_total.counts.data
+                livetime += camera_frame_obs.observation_live_time_duration
 
         return count_map_background, exp_map_background, exp_map_background_total, livetime
```

Comment:
Adding a few commentary could help a easier and faster understanding of the function.

Reply:
I added some comments.